src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java

```diff
@@ -428,8 +429,8 @@ public int rerunBuild(final String externalizableId) {
         LOGGER.fine("Running build again: " + externalizableId); //$NON-NLS-1$
         final AbstractBuild<?, ?> triggerBuild = (AbstractBuild<?, ?>) Run.fromExternalizableId(externalizableId);
         final AbstractProject<?, ?> triggerProject = triggerBuild.getProject();
-        final Future<?> future = triggerProject.scheduleBuild2(triggerProject.getQuietPeriod(), new MyUserIdCause(),
-                removeUserIdCauseActions(triggerBuild.getActions()));
+        final Future<?> future = triggerProject.scheduleBuild2(triggerProject.getQuietPeriod(), null,

```

Comment:
Why is a new MyUserIdCause() not added? I believe this is necessary to set the user id to the current user who clicked the 'retrigger' button. If it is not set, the user will come from the last run, which isn't necessarily true.